pkg/kubectl/history.go

```diff
@@ -256,29 +261,50 @@ func (h *StatefulSetHistoryViewer) ViewHistory(namespace, name string, revision
 	})
 }
 
-// controlledHistories returns all ControllerRevisions controlled by the given DaemonSet
-func controlledHistories(extensions clientextensionsv1beta1.ExtensionsV1beta1Interface, apps clientappsv1beta1.AppsV1beta1Interface, namespace, name string) (*extensionsv1beta1.DaemonSet, []*appsv1beta1.ControllerRevision, error) {
+func retrieveVersionedDaemonSet(extensions clientextensionsv1beta1.ExtensionsV1beta1Interface, namespace, name string) (*extensionsv1beta1.DaemonSet, error) {
 	ds, err := extensions.DaemonSets(namespace).Get(name, metav1.GetOptions{})
 	if err != nil {
-		return nil, nil, fmt.Errorf("failed to retrieve DaemonSet %s: %v", name, err)
+		return nil, fmt.Errorf("failed to retrieve DaemonSet %s: %v", name, err)
+	}
+	return ds, nil
+}
+
+func retrieveVersionedStatefulSet(apps clientappsv1beta1.AppsV1beta1Interface, namespace, name string) (*appsv1beta1.StatefulSet, error) {
+	ss, err := apps.StatefulSets(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve StatefulSet %s: %v", name, err)
 	}
+	return ss, nil
+}
+
+// controlledHistories returns all ControllerRevisions controlled by the given API object
+func controlledHistories(apps clientappsv1beta1.AppsV1beta1Interface, obj runtime.Object) ([]*appsv1beta1.ControllerRevision, error) {
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		return nil, fmt.Errorf("failed to obtain accessor for object %s: %v", accessor.GetName(), err)
+	}
+
 	var result []*appsv1beta1.ControllerRevision
-	selector, err := metav1.LabelSelectorAsSelector(ds.Spec.Selector)
+	labelSelector := metav1.LabelSelector{
+		MatchLabels:      accessor.GetLabels(),
+		MatchExpressions: []metav1.LabelSelectorRequirement{},
```

Comment:
@foxish What is the best way to obtain `Spec.Selector` of `runtime.Object`? It seems that `Accessor` class provides only `MatchLabels`, but not `MatchExpressions`. Hence, i set `MatchExpressions` to be empty in this case.